dtm0/dtm0.h

```diff
@@ -91,6 +91,72 @@
  * @{
  */
 
+#include "fid/fid.h"            /* m0_fid */
+#include "fid/fid_xc.h"         /* m0_fid_xc */
+#include "lib/buf.h"            /* m0_bufs */
+#include "lib/buf_xc.h"         /* m0_bufs_xc */
+#include "lib/types.h"          /* uint64_t */
+#include "xcode/xcode.h"        /* M0_XCA_RECORD */
+
+
+struct m0_dtx0_id {
+	struct m0_fid dti_originator_sdev_fid;
+	uint64_t      dti_timestamp;    /* XXX fix the type */
+} M0_XCA_RECORD M0_XCA_DOMAIN(rpc|be);
+
+
+struct m0_dtx0_participants {
+	uint64_t       dtpa_participants_nr;
+	struct m0_fid *dtpa_participants;
+} M0_XCA_SEQUENCE M0_XCA_DOMAIN(rpc|be);
+
+struct m0_dtx0_descriptor {
+	struct m0_dtx0_id           dtd_id;
+	struct m0_dtx0_participants dtd_participants;
+} M0_XCA_RECORD M0_XCA_DOMAIN(rpc|be);
+
+#define DTID1_F "{" DTS0_F "," FID_F "}"
+#define DTID1_P(__tid) (__tid)->dti_timestamp, FID_P(&(__tid)->dti_originator_sdev_fid)
+enum m0_dtx0_payload_type {
+	M0_DTX0_PAYLOAD_CAS,    /** it's supposed to be handled by CAS */
+	M0_DTX0_PAYLOAD_BLOB,   /**
+				 *  configurable handler.
+				 *  @see m0_dtm0_remach_cfg::dtrc_blob_handler()
+				 */
+} M0_XCA_ENUM M0_XCA_DOMAIN(rpc|be);
+
+struct m0_dtx0_payload {
+	uint32_t         dtp_type M0_XCA_FENUM(m0_dtx0_payload_type);
+	struct m0_bufs   dtp_data;
+} M0_XCA_RECORD M0_XCA_DOMAIN(rpc|be);
+
+struct m0_dtm0_redo {
+	struct m0_dtx0_descriptor dtr_descriptor;
+	struct m0_dtx0_payload    dtr_payload;
+};
+
+struct m0_dtm0_p {
+	struct m0_dtx0_id dtmp_id;
+	struct m0_fid     dtmp_sdev_fid;
+};
```

Comment:
This struct is in-memory only? No need to be transferred on-wire?
I think this one is needed to be sent to on-wire, right?
So we need to add `M0_XCA_RECORD M0_XCA_DOMAIN(rpc)`?

Reply:
I think it will be sent over wire, i will update this in dld branch.